source/common/http/utility.cc

```diff
@@ -693,10 +693,11 @@ Utility::PreparedLocalReplyPtr Utility::prepareLocalReply(const EncodeFunctions&
     response_headers->setReferenceContentType(Headers::get().ContentTypeValues.Grpc);
 
     if (response_headers->getGrpcStatusValue().empty()) {
-      response_headers->setGrpcStatus(std::to_string(
-          enumToInt(local_reply_data.grpc_status_
-                        ? local_reply_data.grpc_status_.value()
-                        : Grpc::Utility::httpToGrpcStatus(enumToInt(response_code)))));
+      response_headers->setGrpcStatus(std::to_string(enumToInt(
```

Comment:
I think you should gate the `setGrpcStatus()` call on the presence of valid gRPC status.

If there is no valid gRPC status set, then `grpc-status` shouldn't be set at all... and I don't believe your current change does that.

Also, I didn't follow Envoy's development for last couple of years, so I'm not sure if this is sufficient change.

Reply:
Understand, there is a mapping between http code and grpc equivalent code. legacy do the mapping in case not set. Just added invalidcode.

Lets see if a maintainer following this module closely think if here is the most suitable place to do it.